File: Graduway.Client/graduway-app/src/app/components/dashboard/services/employees.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import Employee from '../../../Models/employee';

@Injectable()
export class EmployeesService {
  private _http: HttpClient;
  constructor(private http: HttpClient) {
    this._http = http;
  }

  getEmployees(): Observable<Employee[]> {
    return this._http.get<Employee[]>('https://localhost:44313/employees/get');
  }

  addEmployee(employee: Employee): Observable<boolean> {
    return this._http.post<boolean>('https://localhost:44313/employees/add', employee);
  }

  deleteEmployee(employeeId: number): Observable<boolean> {
    return this.http.delete<boolean>(
      `https://localhost:44313/employees/delete-employee?employeeId=${employeeId}`
    );
  }

  getEmployeeById(employeeId: number): Observable<Employee> {
    return this._http.get<Employee>(
      `https://localhost:44313/employees/get-employee?employeeId=${employeeId}`
    );
  }

  updateEmployee(employee: Employee): Observable<boolean> {
    return this._http.put<boolean>(`https://localhost:44313/employees/update-employee`, employee);
  }
}

File: Graduway.API/Graduway.Web/Graduway.BL/Services/EmployeesService.cs
using Graduway.BL.Services.Interfaces;
using Graduway.BL.UnitOfWork.Interfaces;
using Graduway.DAL.Entities;
using System.Collections.Generic;
using System.Linq;

namespace Graduway.BL.Services
{

    public class EmployeesService : IEmployeeService
    {
        private readonly IUnitOfWork _unitOfWork;

        public EmployeesService(IUnitOfWork unitOfWork)
        {
            this._unitOfWork = unitOfWork;
        }

        public IEnumerable<Employee> GetEmployees()
        {
            return _unitOfWork.EmployeesRepository.Get()
                .OrderBy(x=> x.FirstName);
        }

        public void AddEmployee(Employee employee)
        {
            employee.Id = null;
            _unitOfWork.EmployeesRepository.Insert(employee);
            _unitOfWork.Save();
        }

        public Employee GetEmployeeById(int employeeId)
        {
            return _unitOfWork.EmployeesRepository.GetByID(employeeId);
        }

        public void UpdateEmployee(Employee updatedEmployee)
        {
            Employee employee = _unitOfWork.EmployeesRepository.GetByID(updatedEmployee.Id);
            employee.FirstName = updatedEmployee.FirstName;
            employee.LastName = updatedEmployee.LastName;
            employee.Title = updatedEmployee.Title;
            employee.Address = updatedEmployee.Address;
            employee.Department = updatedEmployee.Department;
            _unitOfWork.EmployeesRepository.Update(employee);
            _unitOfWork.Save();
        }

        public void DeleteEmployee(int employeeId)
        {
            _unitOfWork.EmployeesRepository.Delete(employeeId);
            _unitOfWork.Save();
        }
    }
}
